dask_cuda/utils.py

```diff
@@ -649,3 +649,101 @@ def get_gpu_uuid_from_index(device_index=0):
+
+
+def get_worker_config(dask_worker):
+    import dask
+    from distributed.comm import parse_address
+
+    # assume homogenous cluster
+    plugin_vals = dask_worker.plugins.values()
+    ret = {}
+
+    # device and host memory configuration
+    for p in plugin_vals:
+        for v in dir(p):
+            # ignore hidden attrs as well as setup and CPUAffinity
+            if v.startswith("_") or v == "setup" or v == "cores":
+                pass
+            else:
+                ret[v] = getattr(p, v)
+    for mem in [
+        "memory_limit",
+        "memory_pause_fraction",
+        "memory_spill_fraction",
+        "memory_target_fraction",
+    ]:
+        ret[mem] = getattr(dask_worker.memory_manager, mem)
+
+    # jit unspilling set
+    ret["jit-unspill"] = hasattr(dask_worker.data, "spill_on_demand_initialized")
+
+    # get optional device-memory-limit
+    if ret["jit-unspill"]:
+        ret["device-memory-limit"] = dask_worker.data.manager._device_memory_limit
+    else:
+        device_memory_limit = getattr(dask_worker.data, "device_memory_limit", False)
+        if device_memory_limit:
+            ret["device-memory-limit"] = device_memory_limit
+
+    # using ucx ?
+    scheme, loc = parse_address(dask_worker.scheduler.address)
+    ret["protocol"] = scheme
+    if scheme == "ucx":
+        import ucp
+
+        ret["ucx-transports"] = ucp.get_active_transports()
+
+    # comm timeouts
+    ret["distributed.comm.timeouts"] = dask.config.get("distributed.comm.timeouts")
+
+    return ret
+
+
+async def _get_sched_config(client):
+    worker_ttl = await client.run_on_scheduler(
+        lambda dask_scheduler: dask_scheduler.worker_ttl
+    )
+    extensions = list(
+        await client.run_on_scheduler(
+            lambda dask_scheduler: dask_scheduler.extensions.keys()
+        )
+    )
+    ret = {}
+    ret["distributed.scheduler.worker-ttl"] = worker_ttl
+    ret["active-extensions"] = extensions
+
+    return ret
+
+
+async def _get_cluster_configuration(client):
+    worker_config = await client.run(get_worker_config)
+    ret = await _get_sched_config(client)
+
+    # does the cluster have any workers ?
+    if worker_config:
+        w = list(worker_config.values())[0]
+        ret.update(w)
+
+    return ret
+
+
+def get_cluster_configuration(client, table=False):
+    ret = client.sync(
+        _get_cluster_configuration, client=client, asynchronous=client.asynchronous
+    )
+    if not table:
+        return ret
+    else:
+        from rich.console import Console
+        from rich.table import Table
+
+        table = Table(title="Dask Cluster Configuration")
+        table.add_column("Parameter", justify="left", style="red")
+        table.add_column("Value", justify="left", style="green")
+
+        params = sorted(ret.keys())
+        for p in params:
+            table.add_row(p, str(ret[p]))
+        console = Console()
+        console.print(table)
```

Comment:
Maybe make a CLI out of this? I can imagine this causing all sorts of issues if accidentally called from code when `rich` is installed, as it may try to rich-print on the console from arbitrary places.

Reply:
I added a CLI but left the rich code inside of `utils`.  i don't think it will cause any `rich-printing` .  My thought was that folks like @VibhuJawa / users may still want a nicely formatted table from within a live session

Reply:
> I added a CLI

I think a separate CLI (maybe `dask-get-config` or `dask-print-config`) would be better, given that has nothing to do with `dask-cuda-worker`.

> but left the rich code inside of `utils`. i don't think it will cause any `rich-printing` . My thought was that folks like @VibhuJawa / users may still want a nicely formatted table from within a live session

I agree a nicely formatted table is a nice addition to a live session, and maybe this is my ignorance of `rich` but I thought it was a tool to print nicely formatted tables on a terminal, am I mistaken? What if you call that code arbitrarily from any place in the code, say you want to parse that output and feed it to some other code, what happens then (assuming you have `rich` installed)?

Reply:
> I added a CLI but left the rich code inside of utils. i don't think it will cause any rich-printing . My thought was that folks like @VibhuJawa / users may still want a nicely formatted table from within a live session

I like the nicely formatted output .   I guess if we want to feed it into some other code we just stick with the default of `Table=False` .

Reply:
Rich was, I think, originally designed for terminals but notebook support has existed for a 2 years or so:
https://github.com/Textualize/rich/issues/26

If you'd prefer a text only output I can look into the version mismatch output in dask:

```
+-------------+----------+-----------+----------+
| Package     | client   | scheduler | workers  |
+-------------+----------+-----------+----------+
| dask        | 2022.9.2 | 2022.7.1  | 2022.9.2 |
| distributed | 2022.9.2 | 2022.7.1  | 2022.9.2 |
+-------------+----------+-----------+----------+
```

@jacobtomlinson do you know if there are known issues with using rich in a jupyter session ?